eng/WpfArcadeSdk/tools/CodeGen/DesignTimeTextTemplating.targets

```diff
@@ -0,0 +1,98 @@
+<Project>
+  <ItemGroup>
+    <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
+  </ItemGroup>
+  <!-- Including the Microsoft.TextTemplating.targets requires us to manually import Sdk.Targets because
+       it needs to be imported after Sdk.targets. This causes the BuildDependsOn variable,
+       which is modified by the T4 targets, to be overwritten, so the TransformAll target 
+       doesn’t run before the Build target. -->
+  <!-- <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" /> -->
+  <PropertyGroup>
+    <!-- Get the Visual Studio version: -->
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">16.0</VisualStudioVersion>
+    <!-- Keep the next element all on one line: -->
+    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+    <!-- This is what will cause the templates to be transformed when the project is built (default is false) -->
+    <TransformOnBuild>true</TransformOnBuild>
+    <!-- Set to true to force overwriting of read-only output files, e.g. if they're not checked out (default is false) -->
+    <OverwriteReadOnlyOutputFiles>true</OverwriteReadOnlyOutputFiles>
+    <!-- Set to false to transform files even if the output appears to be up-to-date (default is true)  -->
+    <TransformOutOfDateOnly>false</TransformOutOfDateOnly>
+    <NewtonsoftVersion>12.0.2-beta1</NewtonsoftVersion>
+  </PropertyGroup>
+  <Import Project="$(VSToolsPath)\TextTemplating\Microsoft.TextTemplating.targets" />
```

Comment:
I would imagine that  `VSToolsPath` is defined outside of VS IDE, but within msbuild (i.e., when `'$(MSBuildRuntimeType)'!='Core'`)

What happens when `'$(MSBuildRuntimeType)'=='Core'` (for e.g., when invoking build using `dotnet.exe`)? 

Today, our build is not entirely compatible with `dotnet.exe` because of C++ involvement. That said, our C# projects are, in general, compatible with building with `dotnet.exe`. For e.g. `System.Xaml`, `WindowsBase`, `PresentationBuildTasks` etc. should be buildable with `dotnet.exe` (possibly with minor fixes to props/targets) beause they do not rely on C++/CLI or pure-C++ for builds. Will this continue to be true with the inclusion of `Microsoft.TextTemplating.targets` and related MSBuild Tasks?

Reply:
Yeah i'll give this a go, although looks like `dotnet` builds are broken since System.Xaml.csproj is missing.  i notice we get warnings when you build in dotnet-wpf-int about this, but turns out it breaks core builds. i'll fix this up and make sure that it works

Reply:
the only other thing i saw was some targets defined in `ExtendedAssemblyInfo.targets` that are incompatible when building using `dotnet.exe`. I've updated them so that they don't run in that case and the build works